src/main/java/sorald/processor/CastArithmeticOperandProcessor.java

```diff
@@ -79,9 +80,18 @@ private CtTypeReference getExpectedType(CtBinaryOperator ctBinaryOperator) {
 
             CtExecutableReference ctExecutableReference = ctAbstractInvocation.getExecutable();
             if (ctExecutableReference != null && ctExecutableReference.getParameters() != null) {
-                ctTypeReference =
-                        (CtTypeReference)
-                                ctExecutableReference.getParameters().get(indexInInvocation);
+                List<CtTypeReference> parameters = ctExecutableReference.getParameters();
+
+                if (indexInInvocation > parameters.size()
+                        && parameters.get(parameters.size() - 1).isArray()) {
+                    ctTypeReference =
+                            ((CtArrayTypeReferenceImpl) parameters.get(parameters.size() - 1))
+                                    .getComponentType();
+                } else {
+                    ctTypeReference =
+                            (CtTypeReference)
+                                    ctExecutableReference.getParameters().get(indexInInvocation);
+                }
```

Comment:
This could be a method in its own right, would make the `getExpectedType` easier to read.

Reply:
Done.